### PR TITLE
9.6 muharrem weather forecast with Retrofit

### PR DESCRIPTION
**Pull request by Muharrem-sdet**
part 1 with recycler view and live data and dummy forecast list

### CODE REVIEW

File: WeatherForecast/app/src/main/java/ustun/muharrem/weatherforecast/adapters/ForecastListAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import ustun.muharrem.weatherforecast.utilities.DrawableManager.getImageId
+import java.text.SimpleDateFormat
+
+class ForecastListAdapter(private val forecast: Forecast) :
+    RecyclerView.Adapter<ForecastListAdapter.ForecastViewHolder>() {
+
+    class ForecastViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView)
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ForecastViewHolder {
+        val view =
```
Comment (oliver-basoglu) [inline WeatherForecast/app/src/main/java/ustun/muharrem/weatherforecast/adapters/ForecastListAdapter.kt:20]: No need for new lines.

File: WeatherForecast/app/src/main/java/ustun/muharrem/weatherforecast/adapters/ForecastListAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+                getDateText(forecast.data[position].valid_date, position)
+            weather_icon_forecast_item_view.setImageResource(R.drawable::class.java.getImageId(
+                forecast.data[position].weather.icon))
+        }
+    }
+
+    // TODO get rid of suppressLint
+    // TODO try to get the names of the days in LOCAL Language
+    @SuppressLint("SimpleDateFormat")
+    private fun getDateText(valid_date: String, position: Int): String {
```
Comment (oliver-basoglu) [inline WeatherForecast/app/src/main/java/ustun/muharrem/weatherforecast/adapters/ForecastListAdapter.kt:49]: Create another file named DateUtils. make it object (Will act as singleton) . Move this method there

File: WeatherForecast/app/src/main/java/ustun/muharrem/weatherforecast/data/Forecast.kt [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package ustun.muharrem.weatherforecast.data
+
+import android.os.Parcelable
+import kotlinx.android.parcel.Parcelize
+
+@Parcelize
+data class Forecast (
+
+	val data : List<Data>,
```
Comment (oliver-basoglu) [inline WeatherForecast/app/src/main/java/ustun/muharrem/weatherforecast/data/Forecast.kt:9]: Would suggest to rename this to forecastList and the Data class into Forecast

File: WeatherForecast/app/src/main/java/ustun/muharrem/weatherforecast/views/ForecastViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+import retrofit2.Response
+import ustun.muharrem.weatherforecast.data.API_KEY
+import ustun.muharrem.weatherforecast.data.Forecast
+import ustun.muharrem.weatherforecast.network.GetDataService
+import ustun.muharrem.weatherforecast.utilities.RetrofitClient
+
+class ForecastViewModel : ViewModel() {
+
+    private val _forecastLiveData: MutableLiveData<Forecast> =
+        MutableLiveData()
```
Comment (oliver-basoglu) [inline WeatherForecast/app/src/main/java/ustun/muharrem/weatherforecast/views/ForecastViewModel.kt:17]: Refactor: to upper line

File: WeatherForecast/app/src/main/java/ustun/muharrem/weatherforecast/views/ForecastViewModel.kt [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        val callForecast = getDataService?.getForecast(14,43026, API_KEY)
+        callForecast?.enqueue(object : Callback<Forecast> {
+
+            override fun onResponse(call: Call<Forecast>, response: Response<Forecast>) {
+                _forecastLiveData.value = response.body()
+            }
+
+            override fun onFailure(call: Call<Forecast>, t: Throwable) {
+                // TODO add some error messages for the user
+                t.stackTrace
```
Comment (oliver-basoglu) [inline WeatherForecast/app/src/main/java/ustun/muharrem/weatherforecast/views/ForecastViewModel.kt:32]: ?

**Review by oliver-basoglu — CHANGES_REQUESTED** [review]

File: WeatherForecast/app/src/main/java/ustun/muharrem/weatherforecast/views/ForecastViewModel.kt (hunk 4 above)
Comment (Muharrem-sdet) [reply WeatherForecast/app/src/main/java/ustun/muharrem/weatherforecast/views/ForecastViewModel.kt:17]: These happen automatically when I press ctrl + alt + l for formatting my code

File: WeatherForecast/app/src/main/java/ustun/muharrem/weatherforecast/views/ForecastViewModel.kt (hunk 5 above)
Comment (Muharrem-sdet) [reply WeatherForecast/app/src/main/java/ustun/muharrem/weatherforecast/views/ForecastViewModel.kt:32]: It is good to show some info message here to inform the end-user, like "cannot get forecast data" or something like that.

File: WeatherForecast/app/src/main/java/ustun/muharrem/weatherforecast/adapters/ForecastListAdapter.kt (hunk 2 above)
Comment (Muharrem-sdet) [reply WeatherForecast/app/src/main/java/ustun/muharrem/weatherforecast/adapters/ForecastListAdapter.kt:93]: Done

File: WeatherForecast/app/src/main/java/ustun/muharrem/weatherforecast/data/Forecast.kt (hunk 3 above)
Comment (Muharrem-sdet) [reply WeatherForecast/app/src/main/java/ustun/muharrem/weatherforecast/data/Forecast.kt]: Done

File: WeatherForecast/app/src/main/java/ustun/muharrem/weatherforecast/adapters/ForecastListAdapter.kt (hunk 2 above)
Comment (Kryvovyaz) [reply WeatherForecast/app/src/main/java/ustun/muharrem/weatherforecast/adapters/ForecastListAdapter.kt]: I like this method

File: WeatherForecast/app/src/main/java/ustun/muharrem/weatherforecast/adapters/ForecastListAdapter.kt [hunk 6]
```diff
@@ -6,40 +6,84 @@ import android.view.View
...
+    private val OTHER_DAYS = 2
+    override fun getItemViewType(position: Int): Int {
+        return if (position == 0) {
+            TODAY
+        } else {
+            OTHER_DAYS
+        }
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
```
Comment (oliver-basoglu) [inline WeatherForecast/app/src/main/java/ustun/muharrem/weatherforecast/adapters/ForecastListAdapter.kt:31]: Great job.

File: WeatherForecast/app/src/main/java/ustun/muharrem/weatherforecast/adapters/ForecastListAdapter.kt [hunk 7]
```diff
@@ -6,40 +6,84 @@ import android.view.View
...
-                forecast.data[position].weather.description
-            date_forecast_item_view.text =
-                getDateText(forecast.data[position].valid_date, position)
-            weather_icon_forecast_item_view.setImageResource(R.drawable::class.java.getImageId(
-                forecast.data[position].weather.icon))
+    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
+        when (holder) {
+            is ForecastViewHolder -> {
+                holder.itemView.run {
+                    high_temp_forecast_item_view.text =
```
Comment (oliver-basoglu) [inline WeatherForecast/app/src/main/java/ustun/muharrem/weatherforecast/adapters/ForecastListAdapter.kt:53]: If you could move these methods to individual ViewHolder by creating a method named bind(), would be perfect :)

File: WeatherForecast/app/src/main/java/ustun/muharrem/weatherforecast/screens/settings/SettingsFragment.kt [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+        }
+    }
+
+    private fun setNotificationsCheckbox() {
+        activity?.let {
+            notification_settings_checkbox.isChecked = SharedPrefs.getNotificationsSettings(it)
+        }
+    }
+
+    private fun setDegreeViews() {
```
Comment (Kryvovyaz) [inline WeatherForecast/app/src/main/java/ustun/muharrem/weatherforecast/screens/settings/SettingsFragment.kt:87]: great

**Pull request closed without merging**